magda-web-client/src/Components/Dataset/Add/DatasetAddMetadataPage.tsx

```diff
@@ -157,13 +159,13 @@ class NewDataset extends React.Component<Props, State> {
             }
         };
 
-        const nextButtonOnClick = () => {
+        const nextButtonOnClick = async () => {
             if (step === 5) {
                 // --- review page
                 if (config.featureFlags.previewAddDataset) {
-                    this.gotoStep(step + 1);
+                    await this.gotoStep(step + 1);
```

Comment:
It seems like gotoStep doesn't actually need to be async? It `await`s `this.resetError` but that doesn't seem to return a promise anyway :grimacing:

Reply:
The `gotoStep` on add page will call saveState so probably still need `await` event remove it for `this. resetError`. 
I initially make everything into await just to make everything more consistent to avoid bug (because redundant `await` will produce the same result but missing `await` when it's required will generally cause logic error). 
However, here, I actually encounter a weird issue that:
if I remove the await before `this.resetError`, the `moving invalid input ctrl into viewport logic` will not work (seems some timing issue caused by redux action?).
So ends up adding some comments around `this. resetError` to point this out 😢 
If you have a better solution, please let me know~